Kitodo/src/main/java/org/kitodo/production/services/data/LdapServerService.java

```diff
@@ -128,7 +130,24 @@ private String buildUserDN(User inUser) {
         env.put(Context.PROVIDER_URL, ldapServer.getUrl());
         env.put(Context.SECURITY_AUTHENTICATION, "simple");
         env.put(Context.SECURITY_PRINCIPAL, ldapServer.getManagerLogin());
-        env.put(Context.SECURITY_CREDENTIALS, ldapServer.getManagerPassword());
+
+        String managerPassword = ldapServer.getManagerPassword();
+        if (AESUtil.isEncrypted(managerPassword)) {
+            String securitySecret = ConfigCore.getParameterOrDefaultValue(ParameterCore.SECURITY_SECRET);
+
+            if (StringUtils.isBlank(securitySecret)) {
+                logger.error("The security.secret parameter was not configured in kitodo_config.properties file.");
+                return null;
+            }
+
+            try {
+                managerPassword = AESUtil.decrypt(managerPassword, securitySecret);
+            } catch (Exception e) {
+                logger.error(e.getLocalizedMessage(), e);
+                return null;
```

Comment:
Avoid returning `null` value as `null` checks must be done everywhere this method is used.